dask/array/routines.py

```diff
@@ -404,6 +405,71 @@ def ediff1d(ary, to_end=None, to_begin=None):
     return r
 
 
+def _gradient_kernel(f, grad_varargs, grad_kwargs):
+    return np.gradient(f, *grad_varargs, **grad_kwargs)
+
+
+@wraps(np.gradient)
+def gradient(f, *varargs, **kwargs):
+    f = asarray(f)
+
+    if not all([isinstance(e, Number) for e in varargs]):
+        raise NotImplementedError("Only numeric scalar spacings supported.")
+
+    if varargs == ():
+        varargs = (1,)
+    if len(varargs) == 1:
+        varargs = f.ndim * varargs
+    if len(varargs) != f.ndim:
+        raise TypeError(
+            "Spacing must either be a scalar or a scalar per dimension."
+        )
+
+    kwargs["edge_order"] = math.ceil(kwargs.get("edge_order", 1))
+    if kwargs["edge_order"] > 2:
+        raise ValueError("edge_order must be less than or equal to 2.")
+
+    drop_result_list = False
+    axis = kwargs.pop("axis", None)
+    if axis is None:
+        axis = tuple(range(f.ndim))
+    elif isinstance(axis, Number):
+        drop_result_list = True
+        axis = (axis,)
+
+    for e in axis:
+        if not isinstance(e, Integral):
+            raise TypeError("%s, invalid value for axis" % repr(e))
+        if not (-f.ndim <= e < f.ndim):
+            raise ValueError("axis, %s, is out of bounds" % repr(e))
+
+    if len(axis) != len(set(axis)):
+        raise ValueError("duplicate axes not allowed")
+
+    axis = tuple(ax % f.ndim for ax in axis)
+
+    if issubclass(f.dtype.type, (np.bool8, Integral)):
```

Comment:
np.issubdtype ?

Reply:
I think issubclass is slightly saner -- issubdtype deviates from this behavior in some weird deprecated cases.